### PR TITLE
Add cli flag to read from STDIN

### PR DESCRIPTION
**Pull request by nobodywasishere**
This will be useful for IDE stuff as I won't need to write source code to a temp file.

### CODE REVIEW

File: src/ameba/cli/cmd.cr [hunk 1]
```diff
@@ -59,6 +60,7 @@ module Ameba::Cli
     property describe_rule : String?
     property location_to_explain : NamedTuple(file: String, line: Int32, column: Int32)?
     property fail_level : Severity?
+    property stdin_filename : String? = nil
```
Comment (Sija) [inline src/ameba/cli/cmd.cr:63]:
```suggestion
    property stdin_filename : String?
```

**Comment by nobodywasishere** [comment]
Would help implement https://github.com/crystal-ameba/vscode-crystal-ameba/issues/89

**Review by Sija — APPROVED** [review]

**Comment by straight-shoota** [comment]
A common indicator to read from stdin is the filename `-` (for example: `crystal tool format -`).
`--stdin-filename` could only provide an optional name for the stdin file.

**Comment by nobodywasishere** [comment]
All of the crystal tools (except `format`) and `build`/`run` use the `--stdin-filename` flag to read from STDIN, I was matching that functionality.

**Review by veelenga — APPROVED** [review]
lgtm

**Pull request merged**